### ISSUE TITLE
build problem with openssl 1.1.0?

### ISSUE THREAD

**Issue by carlschuurbiers** [MarcAntoine-Arnaud/asdcplib#2]
There seems to be an issue issue with openssl 1.1.0 in Ubuntu 18
I get these errors when buiding:

`M_prng.cpp: In function 'void Kumu::Gen_FIPS_186_Value(const byte_t*, ui32_t, byte_t*, ui32_t)':
KM_prng.cpp:219:10: error: aggregate 'BIGNUM c_2powb' has incomplete type and cannot be defined
   BIGNUM c_2powb, c_2, c_b;
          ^~~~~~~
KM_prng.cpp:219:19: error: aggregate 'BIGNUM c_2' has incomplete type and cannot be defined
   BIGNUM c_2powb, c_2, c_b;
                   ^~~
KM_prng.cpp:219:24: error: aggregate 'BIGNUM c_b' has incomplete type and cannot be defined
   BIGNUM c_2powb, c_2, c_b;
                        ^~~
KM_prng.cpp:220:3: error: 'BN_init' was not declared in this scope
   BN_init(&c_2powb);  BN_init(&c_2);  BN_init(&c_b);
   ^~~~~~~
KM_prng.cpp:220:3: note: suggested alternative: 'BN_print'
   BN_init(&c_2powb);  BN_init(&c_2);  BN_init(&c_b);
   ^~~~~~~
   BN_print
KM_prng.cpp:248:14: error: aggregate 'BIGNUM bn_tmp' has incomplete type and cannot be defined
       BIGNUM bn_tmp, bn_xkey, bn_x_n;
              ^~~~~~
KM_prng.cpp:248:22: error: aggregate 'BIGNUM bn_xkey' has incomplete type and cannot be defined
       BIGNUM bn_tmp, bn_xkey, bn_x_n;
                      ^~~~~~~
KM_prng.cpp:248:31: error: aggregate 'BIGNUM bn_x_n' has incomplete type and cannot be defined
       BIGNUM bn_tmp, bn_xkey, bn_x_n;
                               ^~~~~~
`

is this difficult to fix? Or should one downgrade to openssl 1.0.2?

**Comment by MarcAntoine-Arnaud** [comment]
I don't think so.
I'll to take a look on it soon as possible.

If you want to test it fast, you can use the docker container, available here: https://hub.docker.com/r/arnaudmarcantoine/asdcplib/

**Comment by gorbi79** [comment]
Same issue here with Ubuntu 18.04 while installing digital_cinema_tools @wolfgangw which depends on asdcplib. I think downgrading to openssl 1.0.2 isn't a good option.

**Comment by MarcAntoine-Arnaud** [comment]
Hi,

Is it possible to provide details about the error ? Some logs ?
It can help to see what can be done to solve the problem and keep 1.1.0
release.

Marc-Antoine

**Comment by MarcAntoine-Arnaud** [comment]
A new version is announced, I'll try to check if it will fix this.

**Comment by nicoleneo** [comment]
2.7.18 fixes it "Remove use of BN_init in OpenSSL API".

I have SHA1 problems in both versions even though SHA1 exists in OpenSSL:
```bash
Undefined symbols for architecture x86_64:
  "_SHA1_Final", referenced from:
      digest_file(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&) in asdcp-util.o
  "_SHA1_Init", referenced from:
      digest_file(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&) in asdcp-util.o
  "_SHA1_Update", referenced from:
      digest_file(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&) in asdcp-util.o
```

**Comment by ronhanson** [comment]
Hey @nicoleneo ,

To fix build on macos, you need to give the right cpp flags and add some linking params like crypto .
The following worked for me :
`make CPPFLAGS="-I$(brew --prefix openssl)/include" LDFLAGS="-lcrypto -L$(brew --prefix openssl)/lib"`

I guess the brew install will also work with : 
`CPPFLAGS="-I$(brew --prefix openssl)/include" LDFLAGS="-lcrypto -L$(brew --prefix openssl)/lib" brew install asdcplib`

Passing by : Hey Marc-Antoine! how are you buddy?

**Comment by MarcAntoine-Arnaud** [comment]
Sorry for the delay, I working on other devs ...

@nicoleneo does it works now ?

**Comment by nicoleneo** [comment]
I gave up using on OSX natively  and ran the Docker image instead. My fix #4 has fixed the BIGNUM issues. Please try the latest master.

**Comment by MarcAntoine-Arnaud** [comment]
I have build today on OSX 10.12.6. So I think we can close.
I'm starting to add Travis CI, including Linux and OSX build.
